nsv13/code/modules/overmap/boarding/boarding.dm

```diff
@@ -28,57 +66,64 @@
-		return FALSE //Cut off here to avoid polling people for a spawn that will never work.
-	if(SSovermap_mode.override_ghost_boarders)
-		message_admins("Failed to spawn boarders for [name] due to admin boarding override.")
-		return FALSE //Allows the admins to disable boarders for event rounds
-	var/list/candidates = list()
-	if(player_check < 5)
-		message_admins("KNPC boarder spawning aborted due to insufficient playercounts.")
-		return FALSE //No... just no. I'm not that mean
 
-	//20 or more players? You're allowed "real" boarders.
-	if(player_check >= 20) // Remove the low pop boarder camping
-		candidates = pollCandidatesForMob("Do you want to play as a boarding team member?", ROLE_OPERATIVE, null, ROLE_OPERATIVE, 10 SECONDS, src)
-	//No candidates? Well! Guess you get to deal with some KNPCs :))))))
-	if(!LAZYLEN(candidates))
-		var/list/knpc_types = list()
-		switch(faction_selection)
-			if("syndicate")
-				knpc_types = list(/mob/living/carbon/human/ai_boarder/syndicate/pistol, /mob/living/carbon/human/ai_boarder/syndicate/smg, /mob/living/carbon/human/ai_boarder/syndicate/shotgun)
-			if("pirate")
-				knpc_types = list(/mob/living/carbon/human/ai_boarder/space_pirate/pistol, /mob/living/carbon/human/ai_boarder/space_pirate/auto_pistol)
-			if("greytide")
-				knpc_types = list(/mob/living/carbon/human/ai_boarder/assistant)
-			if("nanotrasen")
-				knpc_types = list(/mob/living/carbon/human/ai_boarder/ert, /mob/living/carbon/human/ai_boarder/ert/commander, /mob/living/carbon/human/ai_boarder/ert/medic, /mob/living/carbon/human/ai_boarder/ert/engineer)
-
-		var/list/possible_spawns = list()
-		for(var/obj/effect/landmark/patrol_node/node in GLOB.landmarks_list)
-			if(node.get_overmap() == src)
-				possible_spawns += node
-		//Can we establish a drop zone?
-		var/turf/LZ = (possible_spawns?.len) ? get_turf(pick(possible_spawns)) : null
-		if(!LZ)
-			message_admins("KNPC boarder spawn aborted. This ship does not support KNPCs (add some patrol nodes!))")
-			return FALSE
-
-		var/obj/structure/closet/supplypod/centcompod/toLaunch = new /obj/structure/closet/supplypod/syndicate_odst
-		var/shippingLane = GLOB.areas_by_type[/area/centcom/supplypod/supplypod_temp_holding]
-		toLaunch.forceMove(shippingLane)
-		for(var/I = 0; I < amount; I++)
-			var/soldier_type = pick(knpc_types)
-			new soldier_type(toLaunch)
-
-		new /obj/effect/pod_landingzone(LZ, toLaunch)
-		return TRUE
+/** Attempts to spawn knpc boarders for this ship.
+ * Arguments:
+ * * amount: Number of boarders to spawn. Will be automatically choosen, based on SSovermap mode difficulty if !amount.
+ * * faction_selection: String for the faction the boarders belong to. See [/obj/structure/overmap/proc/spawn_boarders] for valid values.
+ * Returns TRUE is successful, FALSE otherwise.
+ */
+/obj/structure/overmap/proc/spawn_knpcs(amount = 3, faction_selection="syndicate")
+	if(!amount)
+		amount = CEILING(1 + (SSovermap_mode.mode.difficulty / 2), 1)
```

Comment:
Why are we defining an amount in the base proc if we then have code to handle if there is no amount defined?

Reply:
This provides an interface to pass in a custom amount. This happens if spawn_boarders is called with a custom amount (for instance for adminbus).

Reply:
Yes, however I'm questioning the decision to predefine it to be 3 and then also include code for when its not defined.

Reply:
Just remove amount = 3 and let the catch resolve it.

Reply:
Oh yeah, that's totally a bug. Thanks for catching it.